### PR TITLE
fix embed protocol

### PR DESCRIPTION
**Pull request by pngwn**
Fixes websocket protocol when embedding via spaces.

# Description

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: ui/packages/app/src/api.ts [hunk 1]
```diff
@@ -119,21 +119,22 @@ export const fn =
...
-			var ws_path =
-				is_embed || location.pathname === "/" ? "/" : location.pathname;
-			var ws_host =
-				BUILD_MODE === "dev" || location.origin === "http://localhost:3000"
-					? BACKEND_URL.replace("http://", "").slice(0, -1)
-					: is_embed
-					? new URL(api_endpoint).host
-					: location.host;
-			const WS_ENDPOINT = `${ws_protocol}//${ws_host}${ws_path}queue/join`;
+
```
Comment (freddyaboulton) [inline ui/packages/app/src/api.ts:122]: @pngwn Do we need to do a pre-release to test this?

**Comment by abidlabs** [comment]
Let me do a pre-release to test

**Comment by abidlabs** [comment]
Tested and this new version continues to work for me locally so let's issue an actual release and test on the website (cc @aliabd)

**Comment by pngwn** [comment]
The webcomponent embed worked locally from main, so we somehow need to reproduce the docs error to ensure this _actually_ work. `gradio/timeseries-forecasting-with-prophet` was the embed that is failing on the site.

**Comment by abidlabs** [comment]
I don't understand the error well enough to know how to reproduce it locally. I think the easiest way might be to update the staging website to use `gradio==3.8.1dev1`. Can we try that @aliabd?

**Comment by aliabd** [comment]
I can build staging from this branch, which will be the same as using 3.8.1dev1 (which I could also do).

But we don't need to update the spaces to use that version correct?

**Comment by aliabd** [comment]
live here https://www.gradio-staging.com/docs/

Everything seems to be working fine!

**Pull request merged**